checker/cost.go

```diff
@@ -476,6 +476,9 @@ func (c *coster) sizeEstimate(t AstNode) SizeEstimate {
 	if l := c.estimator.EstimateSize(t); l != nil {
 		return *l
 	}
+	if _, ok := t.Expr().ExprKind.(*exprpb.Expr_CallExpr); ok {
```

Comment:
It appears to me that change causes the default size cost for unhandled functions to be 1. If true I think it is potentially risky.

Consider the case of kubernetes or another developer adding a new function to their custom CEL library. If the developer forgets to implement the new case in their `EstimateSize` function, CEL will now cost their function at `1` instead of `MaxUint64` as before.

I think it might be safer to handle the builtins specifically, and to do so before calling `c.estimator.EstimateSize`, so that it is clear that unimplemented functions have large default cost.

Reply:
A potential alternative would be to check the return type of the call expr here and set it to 1 if the result type is a scalar and MaxInt if it's some variable width type (map, list, string, types).

Reply:
In the worst-case, returning a default value of 1 would be an underestimate. The use of a max value was intended to signal that a function didn't have a cost associated. It seems like associating costs with the builtin operators would also be useful since even equality could require iteration over all list or map entries to determine whether the values are equal and that's certainly not a cost of 1.

Reply:
That's true. Fortunately the code we're working with here is estimating the (min, max) size of the result, not the cost of the function.

That said, I agree that we're not estimating the cost of all functions right. We default to assuming cost 1 of functions unless we write a cost handler for them... Maybe better if we handle that as a separate issue though.

Reply:
Marking this as resolved now that the return type is checked.